support-frontend/test/utils/CheckoutValidationRulesTest.scala

```diff
@@ -134,6 +137,15 @@ class DigitalPackValidationTest extends AnyFlatSpec with Matchers {
     DigitalPackValidation.passes(requestMissingAddressLineAndCity) shouldBe false
   }
 
+  it should "succeed when there is a valid corporate sub" in {
+    val corporateSub = validDigitalPackRequest.copy(
+      product = DigitalPack(GBP, Monthly, Corporate),
+      paymentFields = Right(CorporateRedemption("test-code", "1"))
+    )
+
+    DigitalPackValidation.passes(corporateSub) shouldBe true
```

Comment:
consider testing the other cases, e.g. that corporate does NOT work for GW/paper/etc.

I know we decided against it due to sheer amount of work, but if we could have got it into the types, then we wouldn't need tests as the type system would prove it.

Reply:
I've added a test to check that only digisubs can use corporate redemptions